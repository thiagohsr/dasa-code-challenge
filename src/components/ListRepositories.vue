<template>
  <ul v-bind:class="classes.list">
    <li
      v-bind:class="classes.listItem"
      v-for="item in userRepositories"
      :key="item.id"
    >
      <p>
        <span>{{ item.name }}</span>
        <span v-bind:class="classes.stars">{{ item.stargazers_count }}</span>
      </p>
      <!-- <p>{{ item.description }}</p> -->
    </li>
  </ul>
</template>
<script>
import classes from "@/utils/cssTranspilation";
import bookmarkIcon from "@/assets/svg/bookmark-24px.svg";
import starIcon from "@/assets/svg/stars-24px.svg";

const style = {
  list: {
    backgroundColor: "#FFFFFF",
    borderRadius: 5,
    boxShadow:
      "0px 9px 18px rgba(0, 0, 0, 0.18), 0px 6px 5px rgba(0, 0, 0, 0.24)",
    display: "flex",
    flexDirection: "column",
    listStyle: "none",
    paddingLeft: 0
  },
  listItem: {
    background: {
      image: `url(${bookmarkIcon})`,
      repeat: "no-repeat",
      position: "10px center"
    },
    borderBottomColor: "#f3f3f3",
    borderBottomWidth: 1,
    borderBottomStyle: "solid",
    boxSizing: "border-box",
    "& p": {
      width: "100%",
      display: "inline-flex",
      flexDirection: "row",
      justifyContent: "space-between",
      alignItems: "center"
    },
    paddingRight: 15,
    paddingLeft: 45
  },
  stars: {
    background: {
      image: `url(${starIcon})`,
      repeat: "no-repeat",
      position: "right center"
    },
    color: "#9E9E9E",
    paddingTop: 8,
    paddingLeft: 15,
    paddingRight: 30,
    paddingBottom: 8
  }
};

export default {
  name: "ListRepositories",
  computed: {
    userRepositories() {
      return this.$store.state.userRepositories;
    }
  },
  data() {
    return {
      classes: classes(style)
    };
  },
  props: {}
};
</script>
