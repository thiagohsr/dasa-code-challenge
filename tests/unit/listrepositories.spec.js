import { shallowMount, mount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import ListRepositories from "@/components/ListRepositories.vue";

const localVue = createLocalVue();

localVue.use(Vuex);

describe("ListRepositories", () => {
  let store;
  let state;
  let getters;
  beforeEach(() => {
    state = {
      githubUser: "",
      errorMessage: "",
      userRepositories: [
        {
          html_url: "www",
          name: "name of repository",
          stargazers_count: 123
        },
        {
          html_url: "www2",
          name: "name of repository2",
          stargazers_count: 123
        }
      ],
      paginationLinks: {},
      isLoading: false
    };
    getters = {
      githubUser: state => state.githubUser,
      paginationLinks: state => state.paginationLinks,
      userRepositories: state => state.userRepositories,
      reposCount: state => state.paginationLinks
    };
    store = new Vuex.Store({
      state,
      getters
    });
  });

  it("should render properly ListRepositories view", () => {
    const wrapper = shallowMount(ListRepositories, {
      store,
      localVue,
      computed: {
        userRepositories() {
          return [];
        }
      }
    });
    const html = wrapper.vm.$el;
    expect(wrapper.name()).toBe("ListRepositories");
    expect(html).toMatchSnapshot();
  });

  it("should render properly ListRepositories view with items", () => {
    const wrapper = mount(ListRepositories, {
      store,
      localVue,
      computed: {
        userRepositories() {
          return store.getters.userRepositories;
        }
      }
    });
    const renderedItems = wrapper.findAll("li");
    expect(renderedItems.length).toBe(2);
  });
});
