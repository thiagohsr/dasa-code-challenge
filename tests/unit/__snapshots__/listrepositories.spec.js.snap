// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListRepositories should render properly ListRepositories view 1`] = `
<ul
  class="list-0-0-1"
/>
`;
